### ISSUE TITLE
[Enhancement]: Book rating system / connect with Goodreads 

### ISSUE THREAD

**Issue by scuppasteve** [advplyr/audiobookshelf#701]
### Describe the feature/enhancement

As Audiobookshelf is on the road to supplanting other audiobook players and as a library. I would like to see the option to rate books as they are completed. Either as part of a book finishing playing, where it prompts a rating on the app and saves it to the server. Or you can just go in at any time and rate any books same as you can mark books completed.

A second request would be to have Goodreads rating score listed in the book info, as well as syncing your rating in Audiobookshelf with your Goodreads "completed" books and saving the ratings there.

**Comment by cyturralde** [comment]
I like this but if you wanted to take it a step farther (I don't know if this is possible or not), it would be great to have an implementation that when you are done reading on here it marks it as read on Goodreads as well.

**Comment by scuppasteve** [comment]
> marks it as read on Goodreads as well

Yep, that is what i was hoping as well. I have a collection setup for books i want to read, but if that could be sync'd from Goodreads as well that, would be awesome. Ultimately in the short term i would just like an easy way of recording my rating of a book and i can manually enter it on Goodreads later, i just want to record my main impressions immediately.

**Comment by Jdiesel87** [comment]
Goodreads no longer has a public API

https://github.com/advplyr/audiobookshelf/issues/281

**Comment by kymahi** [comment]
> > Goodreads no longer has a public API
> > #281
> 
> How does Readarr sync back and for books in collection / want to read / etc?

It is still supporting the existing API for those who have keys for it, but at no longer distributing new keys, and there’s talks of what their next steps in regards to a public API is.

https://help.goodreads.com/s/article/Does-Goodreads-support-the-use-of-APIs

**Comment by raviwarrier** [comment]
If not sync with goodreads, can we have an internal rating system. This way, users of the same library can at least see what others think of the book they are considering reading.

**Comment by advplyr** [comment]
Goodreads won't be supported see https://github.com/advplyr/audiobookshelf/issues/281#issuecomment-1159572840

Storing a rating with library items is something to consider with enough support. Another relevant request about ratings is #398

**Comment by Otsira** [comment]
Hi guys. I'm thinking of tackling this proposal as so: Adding a tag with the rate and review in the metadata and the ability to export the reviews as a csv to be imported in [goodreads](https://help.goodreads.com/s/article/How-do-I-import-or-export-my-books-1553870934590). 

Is there anything else I should add?

**Comment by raviwarrier** [comment]
That might work, though, my personal preference is for me to be able to display/view the ratings of the book in ABS. GR can wait or can be ignored since as @advplyr says, it's a closed system and there's no point trying to pursue it. The primary purpose of displaying rating system in ABS is so that all the users in the system can see what others have to say about the book irrespective of whether its exported.

**Comment by dessalines** [comment]
How about using the open source software Bookwyrm for an API?

Example: 

https://bookwyrm.social/book/31229/s/the-vampire-lestat

**Comment by yegods** [comment]
I don't know if there is any progress on this, buy I would like to add my two cents, as it is one of the features I came here to request :)

I would like to see ratings for the books in the ABS library... hopefully in the list views as well as details.

As for Goodreads... even if it is a closed system, plenty of people use it for ratings and reviews, and so maybe just a link to a goodreads item or related search, from the details page may be sufficient?

**Comment by austinphilp** [comment]
I'd like to add a +1 for internal reviews. Would be handy for users on the same server to be able to highlight particularly good books and help to avoid less-than-good ones

**Comment by turnercore** [comment]
I personally like the implementation of a rating system that grabs ratings from some external source (Audible, Goodreads, some Audiobookshelf supported option) to give an initial rating, but allow the user to rate it which will overwrite the rating and show the user's rating (perhaps in a different color). This gives a rating system for books you haven't reviewed yet, but allows a user to record their own rating on top of it to look back and see what they actually liked. (This could also tie in to a recommendation system, if audiobookshelf ever wanted to dive into that)

**Comment by Phonedudeak** [comment]
Just wanted to pop in and voice our +1 for internal reviews for users to rate books in the app.

**Comment by cyrrelsneer** [comment]
I also support the rating request. Would love to see mine but also from other users as well on the same server. Would be really nice

**Comment by basings** [comment]
+1 for a simple 5 star rating system to be able to differentiate between book versions.

**Comment by sandmann91** [comment]
+1 for rating and commenting

**Comment by iconoclasthero** [comment]
I think the DNF concept https://github.com/advplyr/audiobookshelf/issues/2490 is related to this requrest as they both contribute to generally the same information, i.e., the reader's impression of the book.

**Comment by ameeno** [comment]
+1 from me, should be able to have internal ratings, but import ratins from metadata provider such as goodreads or audible :)

**Comment by JohanPotgieter** [comment]
Same here, rating and review is the major feature still missing from ABS

**Comment by Xeth84** [comment]
+1 I would love the ability to import review scores from Goodreads.

**Comment by raviwarrier** [comment]
> Goodreads won't be supported see [#281 (comment)](https://github.com/advplyr/audiobookshelf/issues/281#issuecomment-1159572840)
> 
> Storing a rating with library items is something to consider with enough support. Another relevant request about ratings is [#398](https://github.com/advplyr/audiobookshelf/issues/398)

@advplyr I just came across this project on github. Would this help? https://biblioreads.eu.org/

**Comment by Jdiesel87** [comment]
Hardcover is the developer/consumer friendly alternative to Goodreads. There is already an API and they are working on open sourcing the entire project

https://hardcover.app/

**Comment by Mlitz** [comment]
Implementing Hardcover in all the ways should be end goal. book rating, book progress, metadata, ect.

**Comment by asebbas** [comment]
definitely need some way to store ratings (personal / goodreads / or others). Even custom metadata fields would make me happy
